flink-runtime/src/main/java/org/apache/flink/runtime/scheduler/adaptive/CreatingExecutionGraph.java

```diff
@@ -119,6 +124,26 @@ public void handleGlobalFailure(Throwable cause) {
         context.goToFinished(context.getArchivedExecutionGraph(JobStatus.FAILED, cause));
     }
 
+    private void deployExecutionGraph(ExecutionGraph executionGraph) {
+        for (ExecutionJobVertex executionJobVertex : executionGraph.getVerticesTopologically()) {
+            for (ExecutionVertex executionVertex : executionJobVertex.getTaskVertices()) {
+                deploySafely(executionVertex);
+            }
+        }
+    }
+
+    private void deploySafely(ExecutionVertex executionVertex) {
+        try {
+            executionVertex.deploy();
+        } catch (JobException e) {
+            handleDeploymentFailure(executionVertex, e);
+        }
+    }
+
+    private void handleDeploymentFailure(ExecutionVertex executionVertex, JobException e) {
+        executionVertex.markFailed(e);
+    }
```

Comment:
`ExecutionVertex.markFailed` will trigger an `updateTaskExecutionState` which will only be processed by an `StateWithExecutionGraph`. Hence, I think this will now simply be ignored.